lua/tardis/sh_screens.lua

```diff
@@ -316,139 +460,177 @@ function TARDIS:LoadScreenUI(screen)
 		end
 	end
 	table.SortByMember(screen.screens,1,true)
-	
-	local mmenu = vgui.Create("DPanel",main)
-	mmenu:SetSize(main:GetSize())
-	mmenu:SetPos(0,0)
-	screen.mmenu=mmenu
-	
-	self:LoadButtons(screen,mmenu,function(frame)
+
+	self:LoadButtons(screen, mmenu, function(parent)
 		local buttons={}
 		for k,v in ipairs(screen.screens) do
-			local button=vgui.Create("DButton")
+			local button
+			if TARDIS:GetSetting("visual_gui_enabled")
+			then
+				button = TardisScreenButton:new(parent)
+				button:SetIsToggle(false)
+			else
+				button = vgui.Create("DButton")
+			end
 			button:SetText(v[1])
 			button:SetFont("TARDIS-Default")
 			button.DoClick = function()
-				self:SwitchScreen(screen,v[2])
+				self:SwitchScreen(screen, v[2])
 			end
 			table.insert(buttons,button)
 		end
 		return buttons
-	end)
-	
-	local menubutton=vgui.Create("DButton",titlebar)
-	menubutton:SetSize(titlebar:GetWide()*0.1,titlebar:GetTall())
-	menubutton:SetPos(0,0)
-	menubutton:SetText("Menu")
-	menubutton:SetFont("TARDIS-Default")
-	menubutton.DoClick = function(self)
-		if not ((not IsValid(screen.curscreen)) and mmenu:IsVisible()) then
-			mmenu:SetVisible(not mmenu:IsVisible())
-			pagename:SetText("")
-			if mmenu:IsVisible() and backbutton:IsVisible() then
-				backbutton:SetVisible(false)
-			elseif (not mmenu:IsVisible()) and (#screen.backstack>0) then
-				backbutton:SetVisible(true)
-			end
-		end
-	end
-	screen.menubutton=menubutton
-	
+	end, TARDIS:GetSetting("visual_gui_enabled"))
+
 	return main
 end
 
-function TARDIS:LoadButtons(screen,frame,func)
-	local pages={}
-	local page,spacew,spaceh
-	local function newpage()
-		page=vgui.Create("DPanel",frame)
-		if #pages~=0 then
-			page:SetVisible(false)
+function TARDIS:LoadButtons(screen, frame, func, isvgui)
+	if isvgui ~= nil and isvgui
+	then
+		local layout_rows
+		if screen.is3D2D
+		then
+			layout_rows = math.floor(TARDIS:GetSetting("visual_gui_screen_numrows"))
+		else
+			layout_rows = math.floor(TARDIS:GetSetting("visual_gui_popup_numrows"))
 		end
-		page:SetSize(frame:GetWide()-screen.gap2, frame:GetTall()-(frame:GetTall()*0.1)-screen.gap2)
-		page:SetPos(screen.gap,screen.gap)
-		spacew=page:GetWide()-screen.gap
-		spaceh=page:GetTall()-screen.gap
-		table.insert(pages,page)
-	end
-	local function movebutton(button)
-		local x,y=button:GetSize()
-		local w=page:GetWide()-spacew
-		local h=page:GetTall()-spaceh
-		spacew=page:GetWide()-w-button:GetWide()-screen.gap
-		if spacew < 0 then
-			h=h+button:GetTall()+screen.gap
-			spaceh=page:GetTall()-h
-			spacew=page:GetWide()-screen.gap
-			w=page:GetWide()-spacew
-			spacew=page:GetWide()-w-button:GetWide()-screen.gap
+
+		local layout = HexagonalLayout:new(frame, layout_rows, 0.15)
+
+		for k,v in ipairs(screen.screens) do
+			local button
+			button = TardisScreenButton:new(frame)
+			button:SetIsToggle(false)
+			button:SetText(v[1])
```

Comment:
From testing: Text size is inconsistent on the normal vs big popup size for entries on the menu with no icon

Reply:
Closing this conversation with issue moved to:
https://github.com/MattJeanes/TARDIS/issues/360#issuecomment-785290092